src/common.rs

```diff
@@ -3387,6 +3387,75 @@ impl Cpu {
     pub fn frequency(&self) -> u64 {
         self.inner.frequency()
     }
+
+    /// Returns the CPU's arch (e.g. x86_64, arm64, ...).
+    ///
+    /// ```no_run
+    /// use sysinfo::{System, RefreshKind, CpuRefreshKind};
+    ///
+    /// let s = System::new_with_specifics(
+    ///     RefreshKind::new().with_cpu(CpuRefreshKind::everything()),
+    /// );
+    /// for cpu in s.cpus() {
+    ///     println!("{}", cpu.arch());
+    /// }
+    /// ```
+    pub fn arch(&self) -> CpuArch {
+        self.inner.arch()
+    }
+}
+
+/// Enumeration of CPU architectures
+#[derive(PartialEq, Clone, Debug)]
+pub enum CpuArch {
+    /// `arm` platform
+    ARM,
+    /// `aarch64` or `arm64` platform
+    ARM64,
+    /// `mips` platform
+    MIPS,
+    /// `mips64` platform
+    MIPS64,
+    /// `powerpc` platform
+    POWERPC,
+    /// `powerpc64` platform
+    POWERPC64,
+    /// `riscv32` platform
+    RISCV32,
+    /// `riscv64` platform
+    RISCV64,
+    /// `wasm` platform
+    WASM,
+    /// `wasm64` platform
+    WASM64,
+    /// `x86` platform
+    X86,
+    /// `x64_64` AMD/Intel platform
+    X86_64,
+    /// Unknown platform
+    UNKNOWN,
+}
+
+impl CpuArch {
+    /// Convert an arch given as a string to CpuArch enum
+    pub fn from_string(arch_str: String) -> Self {
```

Comment:
Why not `&str` instead? And implementing `From<str>` directly instead of writing our own.

Reply:
Fixed, implemented ToString in the same time.